### ISSUE TITLE
Android crashes on .stopStreaming()

### ISSUE THREAD

**Issue by rennard** [apivideo/api.video-reactnative-live-stream#34]
**Describe the bug**
Using the example code in the readme, streaming works fine but as soon as I stop recording the app crashes.

**To Reproduce**
1. Use example code
2. Press record button to stop streamin

**Expected behavior**
The app to not crash

**Smartphone (please complete the following information):**
 - Device: Realme 8
 - OS: 12

**Additional context**
I'm using Expo, consistently works for streaming but as soon as I stop it crashes every single time

**Comment by rennard** [comment]
[log.txt](https://github.com/apivideo/api.video-reactnative-live-stream/files/9456176/log.txt)
log file attached

**Comment by rennard** [comment]
```
import React, { useCallback, useEffect, useRef, useState } from 'react';
import { View, TouchableOpacity } from 'react-native';
import { LiveStreamView } from '@api.video/react-native-livestream';

const RecordStream = () => {
  const ref = useRef(null);
  const [streaming, setStreaming] = useState(false);

  return (
    <View style={{ flex: 1, alignItems: 'center', backgroundColor: 'grey' }}>
      <>
        <LiveStreamView
          style={{ flex: 1, backgroundColor: 'black', alignSelf: 'stretch', backgroundColor: 'lightgrey' }}
          ref={ref}
          camera='back'
          video={{
            fps: 30,
            resolution: '720p',
            bitrate: 128000, // # 2 Mbps
          }}
          audio={{
            bitrate: 128000,
            sampleRate: 44100,
            isStereo: true,
          }}
          isMuted={false}
          onConnectionSuccess={() => {
            console.log('success')
          }}
          onConnectionFailed={(e) => {
            console.log('failed')
            console.error(e)
          }}
          onDisconnect={() => {
            console.log('disconnect')
          }}
        />
        <View style={{ position: 'absolute', bottom: 40 }}>
          <TouchableOpacity
            style={{
              borderRadius: 50,
              backgroundColor: streaming ? 'red' : 'white',
              width: 50,
              height: 50,
            }}
            onPress={() => {
              if (streaming) {
                ref.current?.stopStreaming();
                setStreaming(false);
              } else {
                ref.current?.startStreaming(API_KEY_HERE);
                setStreaming(true);
              }
            }}
          />
        </View>
      </>
      {/* )} */}
    </View>
  );
}

export default RecordStream;
```

**Comment by ThibaultBee** [comment]
Hi,

What version of the library do you use?

When you call `startStreaming`, the library creates a new encoder object to be ready for the next live and I don't know why but the system could not create it. Usually it is because the encoding settings are not supported (like resolution is not supported,...). Could you perform the same test with our Android native library (StreamPack)?

For the example code, could you please give a full project instead?

By the way 128000 bps for video is way too low.

**Comment by rennard** [comment]
1.2.0

Is there any way I can determine which parameters are supported? I've been trying to perform the same Android native however I'm not sure your example is up-to-date, I'm trying to implement the `ConnectionChecker` and I can't find it to reference. 

Example code https://drive.google.com/file/d/1z94XDG3r6Cd6VDHiAB_M7LjoSZ70EkdH/view?usp=sharing

Thanks, I've changed that now

**Comment by ThibaultBee** [comment]
> 1.2.0
Ok perfect!

> Is there any way I can determine which parameters are supported?
Not on the RN part yet

> I've been trying to perform the same Android native however I'm not sure your example is up-to-date, I'm trying to implement the `ConnectionChecker` and I can't find it to reference.
Just use the example code or the example from StreamPack.

**Comment by rennard** [comment]
I've just spent several hours trying to implement both your native SDK and StreamPack. I've managed to implement it with your SDK and the same thing happens, I'm struggling to get StreamPack working to be honest as my android experience is limited and the example code leaves some stuff out. I may well be doing something wrong but I'm reaching the limit of my knowledge at this point.

To run my example it requires using expo's eas build, it's basically a yarn install and then use the expo cli to build and install. I can send you a dev build if you like. I should also state it doesn't crash in an Android emulator.

The only remaining thing I can try is to build in a bare react native project

**Comment by rennard** [comment]
Is there any chance you could send me a full StreamPack example to try and run?

**Comment by ThibaultBee** [comment]
Yes, of course, I should have said that before : https://github.com/ThibaultBee/StreamPack/suites/7900508529/artifacts/335986000
If it does not crash on an Android emulator, I am pretty sure it won't crash with my device either. Unfortunately it looks like an issue with this model of phones...

**Comment by rennard** [comment]
Thanks for your reply. I tried with SteamPack and it worked fine on the original device and a new one. I went back to your library to try and it wouldn't work on either device

**Comment by ThibaultBee** [comment]
I ran your example code but  the application did not crash. Is there specific step to reproduce your issue?

**Comment by rennard** [comment]
Strange.. Unfortunately not. It was just a case of starting the stream and then stopping it. I think occasionally it would work but 90% of the time it wouldn't
